packages/broker/test/unit/ConfigWizard.test.ts

```diff
@@ -1,178 +1,202 @@
-                    },
-                    storageNode: '0x31546eEA76F2B2b3C5cC06B1c93601dc35c9D916'
-                }
+        it ('should throw when attempting to mkdir on existing path', () => {
+            try {
+                const parentDirPath = '/home/'
+                createStorageFile(config, {
+                    parentDirPath,
+                    parentDirExists: false,
+                })
+            } catch(e){
+                expect(e.code).toBe('EEXIST')
+                expect(e.syscall).toBe('mkdir')
             }
         })
-    })
 
-    it ('should store the generated config', async () => {
-        mockPromptMethod(wizard, {
-            generateOrImportEthereumPrivateKey: 'generate',
-            selectedItems: ['Websocket', 'MQTT', 'HttpPublish'],
-            destinationFolder: tmpDataDir,
-            overwrite: false,
-            wsPort: 7170,
-            mqttPort: 7171,
-            httpPort: 7172
-        })
-
-        await wizard.generateOrImportPrivateKey()
-        await wizard.selectPlugins()
-        const finalPath = await wizard.storeConfig()
-        expect(existsSync(finalPath)).toEqual(true)
-    })
+        it ('should throw when no permissions on path', () => {
+            try {
+                const parentDirPath = '/home/'
+                const selectDestinationPath = parentDirPath + 'test-config.json'
+                createStorageFile(config, {
+                    selectDestinationPath,
+                    parentDirPath,
+                    fileExists: false,
+                    parentDirExists: true,
+                })
+            } catch(e){
+                expect(e.code).toBe('EACCES')
+                expect(e.syscall).toBe('open')
+            }
+        })
 
-    it ('should store the generated config with overwrite', async () => {
-        mockPromptMethod(wizard, {
-            generateOrImportEthereumPrivateKey: 'generate',
-            selectedItems: ['Websocket', 'MQTT', 'HttpPublish'],
-            destinationFolder: tmpDataDir,
-            overwrite: true,
-            wsPort: 7170,
-            mqttPort: 7171,
-            httpPort: 7172
-        })
-
-        await wizard.generateOrImportPrivateKey()
-        await wizard.selectPlugins()
-        const finalPath = await wizard.storeConfig()
-        expect(existsSync(finalPath)).toEqual(true)
     })
 
-    it ('should return the given destinationFolder when prompted', async() => {
-        mockPromptMethod(wizard, {
-            destinationFolder: tmpDataDir,
-            overwrite: true,
+    describe('getConfigFromAnswers', () => {
+        it ('should exercise the `generate` path', () => {
+            const answers = {
+                generateOrImportEthereumPrivateKey: 'Generate',
+            }
+            const config = getConfigFromAnswers(answers)
+            expect(config.ethereumPrivateKey).toBeDefined()
+            expect(config.ethereumPrivateKey.match(/^(0x)?[a-f0-9]{64}$/)).not.toBe(null)
```

Comment:
Could use Jest's stringMatching (https://jestjs.io/docs/expect#expectstringmatchingstring--regexp) for all regexp checks